File: queuer/src/main/java/com/rndapp/task_feed/views/ProjectViewHolder.java
package com.rndapp.task_feed.views;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.rndapp.task_feed.R;
import com.rndapp.task_feed.listeners.OnProjectClickedListener;
import com.rndapp.task_feed.models.Project;

/**
 * Created by ell on 10/28/16.
 */

public class ProjectViewHolder extends RecyclerView.ViewHolder {
    private TextView tv;
    private Project project;
    private OnProjectClickedListener listener;

    public ProjectViewHolder(View itemView, OnProjectClickedListener listener) {
        super(itemView);

        this.listener = listener;
        tv = (TextView)itemView.findViewById(R.id.tv_project);

        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ProjectViewHolder.this.listener.onProjectClicked(project);
            }
        });
    }

    public void setProject(Project project){
        this.project = project;
        tv.setText(project.getName() + ": " + project.getFirstTaskText());
        tv.setBackgroundColor(project.getColor());
    }
}

File: queuer/src/main/java/com/rndapp/task_feed/views/ProjectViewHolder.kt
package com.rndapp.task_feed.views

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import com.rndapp.task_feed.R
import com.rndapp.task_feed.listeners.OnProjectClickedListener
import com.rndapp.task_feed.models.Project

/**
 * Created by ell on 10/28/16.
 */

class ProjectViewHolder(itemView: View, private val listener: OnProjectClickedListener) : RecyclerView.ViewHolder(itemView) {
    private val tv: TextView
    private var project: Project? = null

    init {
        tv = itemView.findViewById(R.id.tv_project) as TextView

        itemView.setOnClickListener { this@ProjectViewHolder.listener.onProjectClicked(project!!) }
    }

    fun setProject(project: Project) {
        this.project = project
        tv.text = project.name + ": " + project.firstTaskText
        tv.setBackgroundColor(project.color)
    }
}
